### ISSUE TITLE
Feature request: add 'week' to `RelativeFormatOptions`

### ISSUE THREAD

**Issue by alexreardon** [formatjs/react-intl#820]
Currently the `formatRelative` function takes these options:

```js
type RelativeFormatOptions = {
    style?: 'best fit' | 'numeric' = 'best fit',
    units?: 'second' | 'minute' | 'hour' | 'day' | 'month' | 'year',
};
```

My proposal is to extend `units?` to also accept `'week'`.

```js
type RelativeFormatOptions = {
    style?: 'best fit' | 'numeric' = 'best fit',
    units?: 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year',
};
```

This would also include the code to make `'week'` work :)

**Comment by ericf** [comment]
The data exists in CLDR to support this for all languages: https://github.com/unicode-cldr/cldr-dates-full/blob/master/main/en/dateFields.json#L142

**Comment by ericf** [comment]
IIRC this would be the steps involved:

- [ ] Update `intl-relativeformat` to use `formatjs-extract-cldr-data@3.0.0`
- [ ] Update `intl-relativeformat` to use the [`weeks` field](https://github.com/yahoo/intl-relativeformat/blob/master/src/core.js#L23)
- [ ] Update `react-intl` to use `formatjs-extract-cldr-data@3.0.0`
- [ ] Update `<FormattedRelative>` to use [`weeks` field](https://github.com/yahoo/react-intl/blob/master/src/components/relative.js#L20)

**Comment by artursvonda** [comment]
Gonna work on this over next few weeks. A lot of packages need to be updated to enable this.

**Comment by longlho** [comment]
fixed in #1301
